### ISSUE TITLE
pycrypto is not declared as a dependency

### ISSUE THREAD

**Issue by pmuller** [fabric/fabric#1477]
When running "pip install fabric", pycrypto is not installed :
https://github.com/fabric/fabric/blob/master/setup.py#L32-L35

However, Fabric uses it directly :
https://github.com/fabric/fabric/blob/master/fabric/decorators.py#L9

So ImportError is raised when I try to use Fabric without manually installing pycrypto.

Environment:
* Python 2.7.10
* Fabric 1.10.2
* Paramiko 2.0.1

**Comment by pmuller** [comment]
Just realized I have Paramiko 2.0.1 while Fabric's setup.py requires an older version.

So I tried to reinstall Fabric with : "pip install -U 'paramiko<2' fabric"

This time I get pycrypto !

But I don't understand why the "<2" constraint was not followed on my previous installation.

**Comment by bitprophet** [comment]
Related to #1461.

Could swear I saw a ticket or PR about that Crypto import, but I can't find any right now after a brief search. Thanks for surfacing this.

As you noticed, you probably ran into this by using a not-latest Fabric version on install, which was lacking the "Paramiko<2" bit. There's not much we can do to the historical published versions to fix this, obviously :( and the latest versions for the 1.10 and 1.11 lines of course, have the paramiko pin, meaning they shouldn't run into this problem.

So I don't see any good fixes for this, if we added crypto to setup.py it wouldn't do anything that the Paramiko pin isn't already doing.

If you've got suggestions I'm all ears! Going to close as unactionable otherwise.
